containers/hawkular-openshift-security-filter/src/main/java/org/hawkular/openshift/auth/org/hawkular/openshift/namespace/NamespaceOverrideMapper.java

```diff
@@ -0,0 +1,279 @@
+
+        String nodes = System.getProperty("hawkular.metrics.cassandra.nodes", "hawkular-cassandra");
+        Arrays.stream(nodes.split(",")).forEach(clusterBuilder::addContactPoint);
+
+        if (System.getProperty("hawkular.metrics.cassandra.use-ssl") != null && !System.getProperty("hawkular.metrics.cassandra.use-ssl").equals("false")) {
+            SSLOptions sslOptions = null;
+            try {
+                String[] defaultCipherSuites = {"TLS_RSA_WITH_AES_128_CBC_SHA", "TLS_RSA_WITH_AES_256_CBC_SHA"};
+                sslOptions = JdkSSLOptions.builder().withSSLContext(SSLContext.getDefault())
+                        .withCipherSuites(defaultCipherSuites).build();
+                clusterBuilder.withSSL(sslOptions);
+            } catch (NoSuchAlgorithmException e) {
+                throw new RuntimeException("SSL support is required but is not available in the JVM.", e);
+            }
+        }
+
+        Cluster cluster = clusterBuilder.build();
+        cluster.init();
+
+        Session session = cluster.connect();
+
+        return session;
+    }
+
+    /**
+     * Creates a keyspace if it does not already exist
+     * @param keyspace The name of the keyspace
+     */
+    private void ensureKeyspace(String keyspace) {
+        ResultSet resultset = session.execute("SELECT * FROM system_schema.keyspaces WHERE keyspace_name = '" + keyspace + "';");
+        if (!resultset.iterator().hasNext()) {
+            session.execute("CREATE KEYSPACE " + keyspace + " WITH replication = {'class': 'SimpleStrategy', 'replication_factor': '1'}  AND durable_writes = true;");
+        }
+    }
+
+    private boolean hasMappingTable() {
+        ResultSet resultset = session.execute("SELECT table_name FROM system_schema.tables WHERE keyspace_name='openshift_metrics' AND table_name='metrics_mappings';");
+        return resultset.iterator().hasNext();
+    }
+
+    private void createMappingTable() {
+        if (!hasMappingTable()) {
+            session.execute("CREATE TABLE openshift_metrics.metrics_mappings (project_id text PRIMARY KEY, project_name text);");
+        }
+    }
+
+    private Map<String,String> getMappingTable() {
+        Map<String,String> mappings = new HashMap<>();
+        if (hasMappingTable()) {
+            ResultSet resultSet = session.execute("SELECT project_id,project_name FROM openshift_metrics.metrics_mappings;");
+
+            Iterator<Row> ri = resultSet.iterator();
+            while (ri.hasNext()) {
+                Row row = ri.next();
+                mappings.put(row.getString("project_id"), row.getString("project_name"));
+            }
+
+        }
+
+        return mappings;
+    }
+
+    private void writeMappingTable(Map<String,String> mappings) {
+        createMappingTable();
+
+        StringBuilder builder = new StringBuilder();
+        builder.append("BEGIN BATCH ");
+
+        for (Map.Entry<String,String> entry: mappings.entrySet()) {
```

Comment:
Creating the table is not done in the batch statement. And I don't think it can be part of a batch statement. Only the inserts are done via the batch statement. Individual async writes should be used.